marker_api/src/context/map.rs

```diff
@@ -10,9 +10,32 @@ use crate::{
     prelude::{HasNodeId, NodeId},
 };
 
-/// A map, which allows the request of AST nodes by their ids.
+/// A map, which allows the request of AST nodes by their ids. An instance of this
+/// map can be accessed from [`MarkerContext::ast`](super::MarkerContext::ast).
 ///
-/// An instance of this map can be accessed from [`MarkerContext::ast`](super::MarkerContext::ast).
+/// # Availability of Nodes
+///
+/// Rustc, as a compiler and driver for Marker, compiles each crate individually
+/// [^compilation-unit]. This setup means that at a given time, there are only
+/// the AST of one crate available. Therefore, it can happen that the AST node
+/// representation for a valid ID is unavailable.
+///
+/// Generally speaking, it's advised to use the failable variant of the provided
+/// functions and just bail if the expected node is unavailable. The API also
+/// provides some functions that automatically unwrap the requested nodes. These
+/// functions, starting with `unwrap_`, are great for prototyping but should be
+/// used carefully.
+///
+/// FIXME(xFrednet): While the driver might not have the dependency loaded, it
+/// usually has some semantic information about what types and functions are
+/// available. The API should provide some way to request this semantic information
+/// based on the ID. (See: rust-marker/marker#266)
```

Comment:
Maybe we should move this FIXME comment outside of the docs and use `//` for it?

Or maybe remove the `FIXME` part and make it a natural part of the doc comment like this.
```suggestion
/// # Future plans
/// While the driver might not have the dependency loaded, it usually has some semantic information 
/// about what types and functions are available. The API should provide some way to request this 
/// semantic information based on the ID. (See: https://github.com/rust-marker/marker/issues/266).
```
Otherwise `FIXME` sound like a comment that unintentionally percolated to the docs